sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -174,3 +209,25 @@ object JDBCOptions {
   val JDBC_TXN_ISOLATION_LEVEL = newOption("isolationLevel")
   val JDBC_SESSION_INIT_STATEMENT = newOption("sessionInitStatement")
 }
+
+class JdbcOptionsInWrite(
+    @transient override val parameters: CaseInsensitiveMap[String])
+  extends JDBCOptions(parameters) {
+
+  import JDBCOptions._
+
+  def this(parameters: Map[String, String]) = this(CaseInsensitiveMap(parameters))
+
+  def this(url: String, table: String, parameters: Map[String, String]) = {
+    this(CaseInsensitiveMap(parameters ++ Map(
+      JDBCOptions.JDBC_URL -> url,
+      JDBCOptions.JDBC_TABLE_NAME -> table)))
+  }
+
+  require(
+    parameters.get(JDBC_TABLE_NAME).isDefined,
+    s"Option '${JDBCOptions.JDBC_TABLE_NAME}' is required. " +
+      s"Option '${JDBCOptions.JDBC_QUERY_STRING}' is not applicable while writing.")
+
+  val destinationTable = parameters(JDBC_TABLE_NAME)
+}
```

Comment:
Is it bad to change `destinationTable` to `table` for simplicity?

Reply:
@maropu I had it as table and refactored it just before i pushed :-). I will change it back.